lib/internal/bootstrap/pre_execution.js

```diff
@@ -315,9 +330,18 @@ function initializeHeapSnapshotSignalHandlers() {
   require('internal/validators').validateSignalName(signal);
   const { writeHeapSnapshot } = require('v8');
 
-  process.on(signal, () => {
+  function doWriteHeapSnapshot() {
     writeHeapSnapshot();
-  });
+  }
+  process.on(signal, doWriteHeapSnapshot);
+
+  // The code above would add the listener back during deserialization,
+  // if applicable.
+  if (isBuildingSnapshot()) {
+    addSerializeCallback(() => {
+      process.removeListener(signal, doWriteHeapSnapshot);
```

Comment:
Yeah, I noticed that too, but I also realized that it’s going to be quite a journey to fix these up across the code base, so I think we should clean them up in follow-up PRs - I did fix the pre-execution ones because I was running into those when working on the PR so I think they are common enough, whereas the report is less commonly used.

Reply:
SGTM, thank you for the clarification.